adapter_pipelines/Optimus/adapter.wdl

```diff
@@ -0,0 +1,239 @@
+  String dss_url
+  Int retry_seconds
+  Int timeout_seconds
+
+  command <<<
+    python <<CODE
+    from pipeline_tools import dcp_utils
+
+    # Get bundle manifest
+    uuid = '${bundle_uuid}'
+    version = '${bundle_version}'
+    dss_url = '${dss_url}'
+    retry_seconds = ${retry_seconds}
+    timeout_seconds = ${timeout_seconds}
+    print('Getting bundle manifest for id {0}, version {1}'.format(uuid, version))
+    manifest_files = dcp_utils.get_manifest_files(uuid, version, dss_url, timeout_seconds, retry_seconds)
+
+    print('Downloading assay.json')
+    assay_json_uuid = manifest_files['name_to_meta']['assay.json']['uuid']
+    assay_json = dcp_utils.get_file_by_uuid(assay_json_uuid, dss_url)
+
+    # Parse inputs from assay_json and write to fastq_inputs
+    print('Writing fastq inputs to fastq_inputs.tsv')
+    lanes = assay_json['content']['seq']['lanes']
+    r1 = [manifest_files['name_to_meta'][lane['r1']]['url'] for lane in lanes]
+    r2 = [manifest_files['name_to_meta'][lane['r2']]['url'] for lane in lanes]
+    i1 = [manifest_files['name_to_meta'][lane['i1']]['url'] for lane in lanes]
+    fastq_inputs = [list(i) for i in zip(r1, r2, i1)]
+    print(fastq_inputs)
+
+    with open('fastq_inputs.tsv', 'w') as f:
+        for line in fastq_inputs:
+            f.write('\t'.join(line) +'\n')
+
+    print('Writing sample ID to inputs.tsv')
+    sample_id = assay_json['has_input']
+    with open('inputs.tsv', 'w') as f:
+        f.write('{0}'.format(sample_id))
+    print('Wrote input map')
+    CODE
+  >>>
+  runtime {
+    docker: "humancellatlas/pipeline-tools:0.1.5"
+  }
+  output {
+    String sample_id = read_string("inputs.tsv")
+    Array[Array[File]] fastq_inputs = read_tsv("fastq_inputs.tsv")
+  }
+}
+
+task inputs_for_submit {
+    Array[Array[File]] fastq_inputs
+    Array[Object] other_inputs
+
+    command <<<
+      python <<CODE
+      inputs = []
+      print('other inputs')
+      with open('${write_objects(other_inputs)}') as f:
+          keys = f.readline().strip().split('\t')
+          for line in f:
+              values = line.strip().split('\t')
+              input = {}
+              for i, key in enumerate(keys):
+                  input[key] = values[i]
+              print(input)
+              inputs.append(input)
+
+      print('fastq_files')
+      fastq_files = '${sep=", " fastq_inputs}'.replace('/cromwell_root/', 'gs://')
+      inputs.append({"name": "fastq_files", "value": fastq_files})
+
+      print('write inputs.tsv')
+      with open('inputs.tsv', 'w') as f:
+          f.write('name\tvalue\n')
+          for input in inputs:
+              print(input)
+              f.write('{0}\t{1}\n'.format(input['name'], input['value']))
+      print('finished')
+      CODE
+    >>>
+
+    runtime {
+      docker: "humancellatlas/pipeline-tools:0.1.5"
+    }
+
+    output {
+      File inputs = "inputs.tsv"
+    }
+}
+
+task outputs_for_submit {
+    Array[Array[File]] tag_gene_exon_log
```

Comment:
This is related to an Elastc ticket #487 that requires further discussion as a team, but for now, I believe it is good to be merged in.